### PR TITLE
Egurin/fix upload minio dir

### PR DESCRIPTION
**Pull request by GulDilin**
### Purpose
* Fix same logs for all environments in minio
* Improve logging information
* Add retry to each test_case info append and artifacts sending to avoid situation with failed all cases list like here https://rpr.cis.luxoft.com/job/RadeonProRenderMayaPluginManual/4203/Test_20Report/Tahoe/AMD_RXVEGA-Windows-Smoke/Results/Maya/launcher.engine.log
* Fix minio retry

### Effect of change
* Artifacts send to dirs with environment label
* Minio send sleep works
* Log messages have more info
* Avoid situation with failed all cases list

### Technical steps
* Import time to minio_client
* Wrap append test case info to try catch
* Add info about ums to log messages
* Remove str(exception) from log message where next log is traceback (it includes exception info)
* Add env_label to each minio file uploading folder name

### Jenkins Builds
* https://rpr.cis.luxoft.com/job/RadeonProRenderBlender2.8PluginManual/3264/
* https://ums.cistest.luxoft.com/products/5f3b1557929d5171e9d25757/5fe591d28b568a2fa8777fe7/summary
* https://rpr.cis.luxoft.com/job/RadeonProRenderMayaPluginManual/4247/Test_20Report/Tahoe/AMD_RXVEGA-Windows-Smoke/Results/Maya/launcher.engine.log (here also you can see error in one case like in 4203 and results are correct)
* https://ums.cistest.luxoft.com/products/5f3b1557929d5171e9d2575a/5fe591478b568a2fa8777efd/5f3bc463471e5c556272974c/5fe591478b568a2fa8777efe
* minio example for 4247 maya https://minio.cistest.luxoft.com/minio/5f3b1557929d5171e9d2575a/5fe591478b568a2fa8777efd/5f3bc463471e5c556272974c/Windows-AMD_RXVEGA/

### CODE REVIEW

**Review by epozine — COMMENTED** [review]
Добавь новые прогоны пожалуйста

**Comment by GulDilin** [comment]
Добавил ещё один try catch и обновил билды

File: progress_monitor.py [hunk 1]
```diff
@@ -93,30 +98,38 @@ def send_finished_cases(session_dir, suite_name):
...
-                    'render_color_path'])) if is_client else -1
-                print("Upload new image for case: {} and get image id: {}".format(test_case, image_id))
-            case_file_data['image_service_id'] = image_id
-
-        with open(os.path.join(session_dir, suite_name, test_case + '_RPR.json'), 'w') as case_file:
-            json.dump([case_file_data], case_file, indent=4, sort_keys=True)
-
-    transferred_test_cases += list(new_test_cases.keys())
+        try:
+            print('Sending artefacts & images for: {}'.format(test_case))
```
Comment (dm1tryG) [inline progress_monitor.py:108]: Может в лог писать?

File: progress_monitor.py [hunk 2]
```diff
@@ -93,30 +98,38 @@ def send_finished_cases(session_dir, suite_name):
...
+
+                if test_case in new_cases_existence_hashes_info and \
+                        new_cases_existence_hashes_info[test_case] and \
+                        'id' in new_cases_existence_hashes_info[test_case]:
+                    image_id = new_cases_existence_hashes_info[test_case]['id']
+                    print("Use id found by hash for case: {} id: {}".format(test_case, image_id))
+                else:
+                    image_id = is_client.send_image(render_color_full_path(session_dir, suite_name, case_file_data[
+                        'render_color_path'])) if is_client else -1
+                    print("Upload new image for case: {} and get image id: {}".format(test_case, image_id))
```
Comment (dm1tryG) [inline progress_monitor.py:121]: Может в лог писать?

File: progress_monitor.py [hunk 3]
```diff
@@ -93,30 +98,38 @@ def send_finished_cases(session_dir, suite_name):
...
+            print('Sending artefacts & images for: {}'.format(test_case))
+            case_file_path = os.path.join(session_dir, suite_name, test_case + '_RPR.json')
+            with open(case_file_path) as case_file:
+                case_file_data = json.load(case_file)[0]
+
+                if test_case in new_cases_existence_hashes_info and \
+                        new_cases_existence_hashes_info[test_case] and \
+                        'id' in new_cases_existence_hashes_info[test_case]:
+                    image_id = new_cases_existence_hashes_info[test_case]['id']
+                    print("Use id found by hash for case: {} id: {}".format(test_case, image_id))
```
Comment (dm1tryG) [inline progress_monitor.py:117]: Может в лог писать?

**Review by Piromancer — APPROVED** [review]

**Review by Piromancer — APPROVED** [review]

**Review by epozine — APPROVED** [review]

**Pull request merged**